sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/queries/sql/SqlQuery0.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.queries.sql;
+
+import static org.apache.beam.sdk.nexmark.queries.NexmarkQuery.IS_BID;
+
+import org.apache.beam.sdk.coders.BeamRecordCoder;
+import org.apache.beam.sdk.extensions.sql.BeamSql;
+import org.apache.beam.sdk.nexmark.model.Bid;
+import org.apache.beam.sdk.nexmark.model.Event;
+import org.apache.beam.sdk.nexmark.model.sql.ToBeamRecord;
+import org.apache.beam.sdk.nexmark.model.sql.adapter.ModelFieldsAdapters;
+import org.apache.beam.sdk.transforms.Filter;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * Query 0: Pass events through unchanged.
+ *
+ * <p>This measures the overhead of the Beam SQL implementation and test harness like
+ * conversion from Java model classes to Beam records.
+ *
+ * <p>{@link Bid} events are used here at the moment, ås they are most numerous
+ * with default configuration.
+ */
+public class SqlQuery0 extends PTransform<PCollection<Event>, PCollection<BeamRecord>> {
```

Comment:
<!--thread_id:cc_150883553_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:0-->
<!--section:context-quote-->
> **echauchot** wrote:
> Part of the job of original Query0 was to mesure the impact of serialization and metrics to serve as a baseline. SqlQuery0 measures only SQL impact. If the user were to compare SqlQuery0 to Query0 he would compare different things. Is this possible to add `Metrics.counter` and encode/decode roundtrip to SqlQuery0?

<!--section:body-->
I agree that it tests a different thing, but I cannot think of a good way to test exactly same stuff in SQL. 

Java Query0 transform takes an element, runs it though the coder and returns with no change. 

One way to measure the serialization in SQL version would be actually to avoid SQL query execution completely. Instead a version of the existing Java Query 0 can be executed against a PCollection of BeamRecords. This way we can measure the coders for BeamRecords. While this can be a better way of measuring coders impact I am not convinced that this can provide a useful baseline. For example, I don't think a user of SQL can control, or even assume how many of such serializations will be done by any actual query. It's hard to reason about what happens internally in projections, joins, udf calls. Or things like type conversions/casts that happen internally between Beam/Calcite can change the way data is treated.

Another way is to execute an actual SQL query. But then it becomes hard to add arbitrary transform or monitoring code into the query, you can only use what SQL allows to express. You can add a UDF with an arbitrary code, but this again will be testing a different thing from Java Query0, as it doesn't even operate on the same BeamRecord as the input or output of the query.

Do you have an opinion on what approach will be better?

Reply:
Indeed, Query0 and SQLQuery0 are two different things almost impossible to compare their serialization, it's pointless to try to. My point was mainly to add the same Metrics call to the SQL queries. SqlQuery0 could serve as a baseline for the SQL part of nexmark comparing it only to other SQL versions of the queries.

Reply:
Can you add metrics call to the sql query0?